src/d3a/d3a_core/redis_connections/redis_communication.py

```diff
@@ -166,9 +166,9 @@ def _handle_redis_job_metadata(self):
                 self._simulation.stop()
 
         except NoSuchJobError:
-            log.error(f"Redis job {self._simulation_id} cannot be found in the Redis job queue. "
-                      f"get_current_job failed. Continue to operate as a daemon job, "
-                      f"without being monitored by RQ.")
+            raise(f"Redis job {self._simulation_id} cannot be found in the Redis job queue. "
```

Comment:
Hmm, good idea to raise an exception with this message, however it is better to raise an exception of a specific type (e.g. D3AException). Also, please remove the `Continue to operate as a daemon job ... ` part, because it does not continue to operate anymore, the process terminates.

Reply:
Yes, you are right, I changed this without testing.